rubin_scheduler/scheduler/detailers/detailer.py

```diff
@@ -683,6 +688,117 @@ def __call__(self, obsarray, conditions):
         return obsarray
 
 
+class LabelRegionDetailer(BaseDetailer):
+    """Label which region(s) of the footprint we are in.
+
+    Parameters
+    ----------
+    label_array : `np.array`
+        A HEALpix array of strings that are labels for each HEALpix.
+        Default of None uses CurrentAreaMap to load labels.
+    field_for_label : `str`
+        Which ObservationArray field should be modified.
+        Default "target_name".
+    camera : `str`
+        Which camera model to use to convert a pointing to
+        HEALpix IDs. Default "LSST".
+    append : `bool`
+        Should the labels be appended to `field_for_label`
+        (True, default), or clobber (False).
+    """
+
+    def __init__(self, label_array=None, field_for_label="target_name", camera="LSST", append=True):
+        if label_array is None:
+            sky = CurrentAreaMap()
+            _footprints, self.label_array = sky.return_maps()
+        else:
+            self.label_array = label_array
+        self.field_for_label = field_for_label
+        nside = hp.npix2nside(np.size(self.label_array))
+        self.append = append
+
+        if camera == "LSST":
+            self.pointing2hpindx = HpInLsstFov(nside=nside)
+        else:
+            raise ValueError("Unknown camera")
+
+    def __call__(self, obs_array, conditions):
+        for i in np.arange(obs_array.size):
+            indx = self.pointing2hpindx(obs_array["RA"][i], obs_array["dec"][i])
+            labels = np.unique(self.label_array[indx])
+            # ignore things that are out of bounds
+            labels = labels[np.where(labels != "")]
+            # If we overlap multiple regions
+            if np.size(labels) > 1:
+                result = ", ".join(labels)
+            else:
+                result = labels[0]
+
+            if self.append:
```

Comment:
How would append come to be used? Isn't an observation only proposed from a single survey, so it should only go through the detailer once?

Reply:
There can be multiple detailers (and they survey itself) setting the `target_name`. So a ToO in a DDF could end up with a target name like `SSO_night_i1, lowdust, EDFS_a`.